### DISCUSSION TITLE
Nextjs 13: Server component fetches data calls twice on build (it's not react strict mode)

### DISCUSSION THREAD

**Discussion by rgioeli** [vercel/next.js#46280] [Help]
Many people have had this issue on stackoverflow and other forums. I've wrote about this here in the discussion before and just had comments about people who agree with me that they're having the same issue and have gone back to Nextjs 12. 

 **Note: make sure you have used yarn build or npm build to replicate this. This is not an issue in dev mode**

My issue is simple to replicate... just create a new Nextjs 13 project with `npx create-next-app@latest` and in **page.js** simply fetch any data like this (I am using jsonplaceholder as an easy way to test this).

```
async function getTodo() {
  const req = await fetch("https://jsonplaceholder.typicode.com/todos/1", {
    next: { revalidate: 10 },
  });
  console.log("I made the call!");
  return await req.json();
}

export default async function Home() {
  const todo = await getTodo();

  return <main>{todo && <p>{todo.title}</p>}</main>;
}
```
Reponse in terminal:

**I made the call!
I made the call!**

After the 10 second revalidation it prints out the log twice again.

What is going on?

**Comment by mattrmedia** [comment]
@rgioeli The reason this is happening is because React is rendering in development with `React.StrictMode`. It's printing the log twice because React is rendering twice and identifying a possible side effect of your code. You have defined your function outside the consuming `Home` component. Try defining and running your async code in a `useEffect` and you should only get your async call logging once. Here is an [example](https://codesandbox.io/p/sandbox/loving-matsumoto-hl1t1o?file=%2Fpages%2Findex.tsx&selection=%5B%7B%22endColumn%22%3A44%2C%22endLineNumber%22%3A10%2C%22startColumn%22%3A44%2C%22startLineNumber%22%3A10%7D%5D). Hope this helps.
